src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplication.groovy

```diff
@@ -24,11 +24,11 @@ class DockerJavaApplication {
     String maintainer = System.getProperty('user.name')
     @Deprecated
     Integer port = 8080
-    Set<Integer> ports = []
+    Set<Integer> ports
     String tag
 
     Integer[] getPorts() {
-        return ports.size() > 0 ? ports : [port]
+        return ports != null ? ports : [port]
```

Comment:
Elvis operator?

```
Integer[] getPorts() {
    ports ?: [port]
}
```

Reply:
Elvis doesn't help here, because the empty list is falsy.